lib/alerts_viewer_web/live/put_flags_in_assigns_hook.ex

```diff
@@ -0,0 +1,24 @@
+defmodule AlertsViewerWeb.PutFlagsInAssignsHook do
+  @moduledoc """
+    Grabs Laboratory flags from session and makes them available as assigns
+  """
+
+  import Phoenix.Component
+
+  def on_mount(:default, _params, session, socket) do
```

Comment:
suggestion: Unless it would be particularly tricky or something, it might be nice to write a quick test for this functionality. I think it would be nice to demonstrate what it does, and would also help guard against regressions.

Reply:
I wasn't able to test this as directly as I wanted, but I got a test around the flag behavior at least. If this hook isn't working and the liveview page can't find an assign, it crashes, so we also have that assurance.